src/plugins.ts

```diff
@@ -0,0 +1,28 @@
+import { Rule } from './prefixer'
+
+export interface RuleSet {
+  selector: string,
+  rules: Rule[]
+}
+
+type Plugin = (set: RuleSet) => RuleSet
+
+const plugins: Plugin[] = []
+
+export function use(plugin: Plugin): void {
+  plugins.unshift(plugin)
+}
+
+export function apply(set: RuleSet) {
+  let newSet = {...set}
```

Comment:
Seems like we don't need this spread. Only if there are plugins will it matter, and we are already shallowly copying the object for each plugin execution.

```suggestion
  let newSet = set
```